File: msme_intel-main/backend/app/models/automation.py
from pydantic import BaseModel
from typing import Optional, List, Dict, Any
from datetime import datetime

class SectorAutomationNeedBase(BaseModel):
    sector_id: str
    regional_context_id: Optional[str] = None
    title: str
    slug: str
    description: Optional[str] = None
    automation_type: str # process_automation, predictive_analytics, etc.
    ceed_quadrant: str # core, expansion, efficiency, disruption
    impact_level: str # transformative, high, medium, incremental
    estimated_roi_percentage: Optional[int] = 0
    implementation_complexity: Optional[str] = "medium"
    time_to_value_weeks: Optional[int] = 0
    target_company_size: Optional[List[str]] = []
    target_employee_range: Optional[str] = None
    target_revenue_range: Optional[str] = None
    prerequisite_tech_maturity: Optional[str] = None
    tech_stack_suggested: Optional[Dict[str, Any]] = {}
    data_requirements: Optional[Dict[str, Any]] = {}
    research_citations: Optional[List[Dict[str, Any]]] = []
    industry_benchmarks: Optional[Dict[str, Any]] = {}
    status: Optional[str] = "draft"
    is_featured: Optional[bool] = False
    priority_rank: Optional[int] = 0
    tags: Optional[List[str]] = []

class SectorAutomationNeed(SectorAutomationNeedBase):
    id: str
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

class AutomationInterestCreate(BaseModel):
    current_pain_points: List[str]
    current_process_description: str
    desired_outcomes: List[str]
    budget_range: Optional[str] = None
    timeline_preference: Optional[str] = None
    existing_systems: Optional[Dict[str, Any]] = {}
    data_availability: Optional[Dict[str, Any]] = {}
    constraints: Optional[List[str]] = []
    success_metrics: Optional[Dict[str, Any]] = {}
    poc_name: str
    poc_email: str
    poc_phone: Optional[str] = None
    poc_designation: Optional[str] = None
    additional_context: Optional[str] = None

class CompanyAutomationInterest(BaseModel):
    id: str
    company_id: str
    automation_need_id: str
    status: str
    current_pain_points: List[str]
    current_process_description: str
    desired_outcomes: List[str]
    budget_range: Optional[str] = None
    timeline_preference: Optional[str] = None
    existing_systems: Optional[Dict[str, Any]] = {}
    data_availability: Optional[Dict[str, Any]] = {}
    constraints: Optional[List[str]] = []
    success_metrics: Optional[Dict[str, Any]] = {}
    poc_name: str
    poc_email: str
    poc_phone: Optional[str] = None
    poc_designation: Optional[str] = None
    additional_context: Optional[str] = None
    ice_approval_status: str
    ice_comments: Optional[str] = None
    interested_at: datetime
    context_added_at: Optional[datetime] = None
    updated_at: datetime

    class Config:
        from_attributes = True

File: msins-main/src/types/automation.ts
export type AutomationType = 
  | 'process_automation'
  | 'predictive_analytics'
  | 'computer_vision'
  | 'nlp_automation'
  | 'iot_integration'
  | 'supply_chain_automation'
  | 'customer_automation'
  | 'financial_automation';

export type AutomationInterestStatus = 
  | 'interested'
  | 'context_added'
  | 'ice_review'
  | 'approved'
  | 'rejected'
  | 'revision_requested';

export interface SectorAutomationNeed {
  id: string;
  sector_id: string;
  regional_context_id?: string;
  
  title: string;
  slug: string;
  description: string;
  
  automation_type: AutomationType;
  ceed_quadrant: 'core' | 'expansion' | 'efficiency' | 'disruption';
  
  impact_level: 'transformative' | 'high' | 'medium' | 'incremental';
  estimated_roi_percentage: number;
  implementation_complexity: 'low' | 'medium' | 'high' | 'very_high';
  time_to_value_weeks: number;
  
  target_company_size: string[];
  target_employee_range: string;
  target_revenue_range: string;
  prerequisite_tech_maturity: string;
  
  tech_stack_suggested: TechStackSuggestion;
  data_requirements: DataRequirements;
  research_citations: ResearchCitation[];
  industry_benchmarks: IndustryBenchmark;
  
  status: 'draft' | 'research' | 'validated' | 'published' | 'archived';
  is_featured: boolean;
  priority_rank: number;
  tags: string[];
  discovery_insight?: {
    session_id: string;
    rationale: string;
    confidence: number;
    quote?: string;
  };
}

export interface TechStackSuggestion {
  primary: string[];
  secondary: string[];
  edge?: string[];
}

export interface DataRequirements {
  structured?: string[];
  unstructured?: string[];
  real_time?: string[];
}

export interface ResearchCitation {
  title: string;
  source: string;
  url?: string;
  year?: number;
}

export interface IndustryBenchmark {
  metric: string;
  current_avg: string;
  potential_improvement: string;
}

export interface CompanyAutomationInterest {
  id: string;
  company_id: string;
  automation_need_id: string;
  
  status: AutomationInterestStatus;
  
  current_pain_points: string[];
  current_process_description: string;
  desired_outcomes: string[];
  budget_range: string;
  timeline_preference: string;
  
  existing_systems: Record<string, any>;
  data_availability: Record<string, any>;
  constraints: string[];
  success_metrics: Record<string, any>;
  
  poc_name: string;
  poc_email: string;
  poc_phone: string;
  poc_designation: string;
  
  additional_context: string;
  
  ice_approval_status: 'pending' | 'approved' | 'rejected' | 'revision_requested';
  ice_comments?: string;
  
  interested_at: string;
  context_added_at?: string;
}
